src/odemis/util/test/util_test.py

```diff
@@ -396,54 +401,255 @@ def test_line_clipping(self):
+        testing.start_backend(ENZEL_CONFIG)
         # it should still be enzel.
         role = model.getMicroscope().role
         self.assertEqual(role, "enzel")
 
     def test_running_backend_different_from_requested(self):
         # run sparc backend
-        test.start_backend(SPARC_CONFIG)
+        testing.start_backend(SPARC_CONFIG)
         # check if the role is sparc
         role = model.getMicroscope().role
         self.assertEqual(role, "sparc")
         # now run another backend (enzel)
-        test.start_backend(ENZEL_CONFIG)
+        testing.start_backend(ENZEL_CONFIG)
         # check if the role now is enzel instead of sparc
         role = model.getMicroscope().role
         self.assertEqual(role, "enzel")
 
 
+class FindClosestTestCase(unittest.TestCase):
+
+    def test_simple(self):
+        """ Finding the closest point in a single list """
+        closest = util.find_closest(15, [1, 2, 3])
+        self.assertEqual(closest, 3, "Failed to get correct output.")
+
+    def test_multiple(self):
+        """ Test finding the closest point in multiple lists """
+        in_exp = {
+            tuple([-1, -5, -6]): -1,
+            tuple([1, 2, 3]): 3,
+            tuple([0, 5, 10]): 5,
+            tuple([-15, 30, -60]): -15
+        }
+
+        for testlist, hardcoded_closest in in_exp.items():
+            closest = util.find_closest(5, testlist)
+            self.assertEqual(hardcoded_closest, closest, "Failed to get correct output for "
+                                                         "the closest value to 5 in %s" % in_exp)
+
+
+class IndexClosestTestCase(unittest.TestCase):
+
+    def test_simple(self):
+        """ Test finding the index of the closest point in a list """
+        simple_list = [1, 5, 6]
+        closest_index = util.index_closest(1, simple_list)
+        self.assertEqual(closest_index, 0,  "Failed to get correct output for "
+                                            "the closest index to 1 in %s" % simple_list)
+
+    def test_multiple(self):
+        """ Test finding the index of the closest point in multiple lists """
+        in_exp = {
+            tuple([-1, -5, -6]): 0,
+            tuple([1, 2, 3]): 2,
+            tuple([0, 5, 10]): 1,
+            tuple([-15, 30, -60]): 0
+        }
+
+        for testlist, hardcoded_closest_index in in_exp.items():
+            closest = util.index_closest(5, testlist)
+            self.assertEqual(hardcoded_closest_index, closest,  "Failed to get correct output for "
+                                                                "the closest index to 5 in %s" % in_exp)
+
+
+class NormalizeRectTestCase(unittest.TestCase):
+
+    def test_already_normalized(self):
+        """ Test normalizing a rectangle with already normalized values """
+        rect = (1, 2, 3, 4)
+        normalized_rect = util.normalize_rect(rect)
+        self.assertEqual(normalized_rect, (1, 2, 3, 4), "Failed to normalize the rectangle")
```

Comment:
You could replace  `(1, 2, 3, 4)` with `rect`, that makes it clear that the output is the same as the input